src/lib/components/ResourceTypeField.js

```diff
@@ -23,111 +20,73 @@ export class ResourceTypeField extends Component {
     return null;
   };
 
+  /**
+   * Generate label value
+   *
+   * @param {object} option - back-end option
+   * @returns {string} label
+   */
+  _label = (option) => {
+    return (
+      option.type_name + (option.subtype_name ? (" / " + option.subtype_name) : "")
+    );
+  }
+
+  /**
+   * Convert back-end options to front-end options.
+   *
+   * @param {array} propsOptions - back-end options
+   * @returns {array} front-end options
+   */
   createOptions = (propsOptions) => {
-    let options = [];
-    for (const type of propsOptions.type) {
-      const subtypes = propsOptions.subtype.filter(
-        (e) => e['parent-value'] === type.value
-      );
+    let options = propsOptions.slice();
 
-      if (subtypes.length > 0) {
-        // Push options corresponding to each subtype
-        options.push(
-          ...subtypes.map((subtype) => ({
-            // NOTE: In this version of semantic-ui-react (0.88.2),
-            // using icon key doesn't show icon in selected text. This does:
-            text: <><i className={type.icon + " icon"}></i><span className="text">{subtype["parent-text"] + " / " + subtype.text}</span></>,
-            value: subtype["parent-value"] + "/" + subtype.value,
-            values: {type: type.value, subtype: subtype.value}
-          }))
-        );
+    // Sort by label
+    options.sort((o1, o2) => {
+      let label1 = this._label(o1);
+      let label2 = this._label(o2);
+      if (label1 < label2) {
+        return -1;
+      } else if (label1 > label2) {
+        return 1;
       } else {
-        // Push an option corresponding to the type only
-        options.push({
-            icon: type.icon,
-            text: type.text,
-            value: type.value,
-            values: {type: type.value}
-        })
+        return 0;
       }
-    }
-    return options;
-  }
+    });
 
-  loadValue(values, fieldPath) {
-    const resourceType = _get(values, fieldPath);
-    let value;
-    if (resourceType && resourceType.type) {
-      value = resourceType.type + (resourceType.subtype ? "/" + resourceType.subtype : "");
-    } else {
-      value = "";
-    }
-    return value;
+    // Convert to front-end format
+    options = options.map(
+      (o) => {
+        const label = this._label(o);
+        return {
+          text: <><i className={o.icon + " icon"}></i><span className="text">{label}</span></>,
+          value: o.id,
+          //         values: {type: type.value}
+        }
+      }
+    );
+    return options;
   }
 
-  renderResourceTypeField = ({ field, form }) => {
-    const {
+  render() {
+    let {
```

Comment:
Please use `const` whenever possible and not `let`.
```suggestion
    const {
```